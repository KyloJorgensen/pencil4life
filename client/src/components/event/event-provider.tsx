'use strict';

import * as React from 'react';
import * as querystring from 'querystring';
import * as moment from 'moment';
import eventListItem from './event-list-item';
import { match } from 'react-router';

export interface IEventContext extends IEventWrapperState, IEventWrapperMethods {
    eventItem?: EventItem;
}

const Context: IEventContext =  {
    getEventItems: (query) => {
        console.error('Error: Cannot getEventItems no EventContext.Provider element in parents');
    },
    getEventItem: (_eventItemId) => {
        console.log('Error: Cannot getEventItem no EventContext.Provider element in parents');
    },
    addEventItem: (params, callback) => {
        console.log('Error: Cannot addEventItem no EventContext.Provider element in parents');
    },
    eventItemList: {},
    eventItems: [],
    page: 0,
    totalPages: 0,
    limit: 0,
    total: 0,
    pageTotal: 0,
}

export const EventContext = React.createContext(Context);

export interface IEventWrapperProps {
    initialState?: IEventWrapperState;
}

export interface IEventWrapperState {
    eventItemList: {};
    eventItems: string[];
    page: number;
    totalPages: number;
    limit: number;
    total: number;
    pageTotal: number;
}

export interface getEventItemsQuery {
    limit?: number,
    format?: 'json',
    discontinued?: boolean; 
}

export interface EventItem extends addEventItemParams {
    _eventItemId: string,
    createdUpdatedDateTime: string,
    discontinued: boolean,
}

export interface addEventItemParams {
    title: string; 
    start_date: string | moment.Moment | Date; 
    end_date: string | moment.Moment | Date;
    details: string;
}

export interface IEventWrapperMethods {
    getEventItems: (query: getEventItemsQuery) => void;
    getEventItem: (_eventItemId: string) => void;
    addEventItem: (params: addEventItemParams , callback: (error, _eventItemId) => void) => void;
}

export class EventWrapper extends React.Component<IEventWrapperProps, IEventWrapperState, IEventWrapperMethods> implements IEventWrapperMethods {
    public static defaultProps: (IEventWrapperProps) = {
        initialState: {
            eventItemList: {},
            eventItems: [],
            page: 0,
            totalPages: 0,
            limit: 0,
            total: 0,
            pageTotal: 0,
        }
    } 
    constructor(props) {
        super(props);
        this.state = this.props.initialState;
        this.getEventItems = this.getEventItems.bind(this);
        this.getEventItem = this.getEventItem.bind(this);
        this.addEventItem = this.addEventItem.bind(this);
    }

    getEventItems(query) {
        let querydefaults: getEventItemsQuery = {
            limit: 100,
            format: 'json',
        };
        if (query) {
            if (!query.discontinued) {
                querydefaults.discontinued = false;
            }
            if (query.limit) {
                querydefaults.limit = query.limit;
            }
        } else {
            querydefaults.discontinued = false;
        }

        let _query = querystring.stringify(querydefaults);
        _query = _query ? '?'+_query : '';

        const request = new Request('/api/event/' + _query, {
            method: 'GET',
            credentials: 'same-origin',
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
            },
        });

        fetch(request)
        .then((response) => {
            return response.json();
        })
        .then((body) => {
            this.setState(() => {
                return {
                    page: body.page,
                    totalPages: body.totalPages,
                    limit: body.limit,
                    total: body.total,
                    pageTotal: body.pageTotal,
                    eventItems: body.eventItems,
                };
            });
        })
        .catch((error) => {
            this.setState(() => {
                return {
                    eventItems: null,
                    page: null,
                    totalPages: null,
                    limit: null,
                    total: null,
                    pageTotal: null,
                };
            });
        });
    }

    getEventItem(_eventItemId) {
        let query = querystring.stringify({
            field: [
                '_id',
                'title',
                'start_date',
                'end_date',
                'details',
                'discontinued',
            ],
            format: 'json',
        });
        query = query ? '?'+query : '';

        const request = new Request('/api/event/' + _eventItemId  + query, {
            method: 'GET',
            credentials: 'same-origin',
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
            },
        });
        
        fetch(request).then((response) => {
            if (response.status < 200 || response.status >= 300) {
                const error = new Error(response.statusText)
                error.message = String(response);
                throw error;
            }
            return response;
        })
        .then((response) => {
            return response.json();
        })
        .then((body) => {
            const {_id, title, start_date, end_date, details, createdAt, updatedAt, discontinued} = body;

            this.setState((preState) => {
                const { eventItemList } = preState;
                    
                let createDateTime = moment(createdAt);
                let updatedDateTime = moment(updatedAt);

                let daymessages = {
                    sameDay: '[Today]',
                    nextDay: '[Tomorrow]',
                    nextWeek: 'dddd',
                    lastDay: '[Yesterday]',
                    lastWeek: '[Last] dddd',
                    sameElse: 'MM/DD/YYYY',
                };
                let createdDay = createDateTime.calendar(null, daymessages);
                let updatedDay = updatedDateTime.calendar(null, daymessages);

                let createdTime = createDateTime.format("h:mm a");
                let updatedTime = updatedDateTime.format("h:mm a");

                let createdUpdatedDateTime = 'Created ';
                if (createdDay !== 'Today' && createdDay !== 'Yesterday') {
                    createdUpdatedDateTime += 'on '
                }
                createdUpdatedDateTime +=  createdDay + ' at ' + createdTime;

                if (createdDay !== updatedDay) {
                    createdUpdatedDateTime += ' Updated '
                    if (updatedDay !== 'Today' && updatedDay !== 'Yesterday') {
                        createdUpdatedDateTime += 'on ';
                    }
                    createdUpdatedDateTime += updatedDay + ' at ' + updatedTime;

                } else if (createdTime !== updatedTime) {
                    createdUpdatedDateTime += ' Updated at ' + updatedTime;
                } 
                
                eventItemList[_id] = {
                    _eventItemId: _id,
                    title: title,
                    start_date: start_date,
                    end_date: end_date,
                    details: details,
                    createdUpdatedDateTime: createdUpdatedDateTime,
                    discontinued: discontinued,
                };

                return {eventItemList: eventItemList};
            });
        })
        .catch((error) => {
            console.log(error);
        });
    };

    addEventItem(params, callback) {
        const payload = {
            title: params.title,
            start_date: params.start_date,
            end_date: params.end_date,
            details: params.details,
        };
            
        const url = '/api/event';
        const request = new Request('/api/event/', {
            method: 'POST',
            credentials: 'same-origin',
            headers: {
                'Content-Type': 'application/json',
                'Accept': 'application/json',
            },
            body: JSON.stringify(payload), 
        });
        fetch(request)
        .then((response) => {
            if (response.status < 200 || response.status >= 300) {
                const error = new Error(response.statusText)
                error.message = String(response);
                throw error;
            }
            return response;
        })
        .then((response) => {
            return response.json();
        })
        .then((response) => {
            callback(false, response._id);
        })
        .catch((error) => {
            callback(true);
        });
    }

    render() {
        const { getEventItems, getEventItem, addEventItem } = this;
        const { eventItems, page, pageTotal, totalPages, limit, total, eventItemList } = this.state;

        const context = {
            eventItems: eventItems,
            eventItemList: eventItemList,
            page: page,
            pageTotal: pageTotal,
            totalPages: totalPages,
            limit: limit,
            total: total,
            getEventItems: getEventItems,
            getEventItem: getEventItem,
            addEventItem: addEventItem,
        }
        
        return (
            <EventContext.Provider value={context}>
                {this.props.children}
            </EventContext.Provider>
        );
    }
}

export default EventWrapper;



export interface eventProviderOptions {
  props?: IEventWrapperProps;
}

export const eventProvider = (Component, options?: eventProviderOptions) => {
  class EventProvider extends React.Component<null, null> {
    render () {
      let eventWrapperProps: IEventWrapperProps = {};
      if (options) {
        if ('props' in options) {
          eventWrapperProps = options.props;
        }
      }
      return (
        <EventWrapper {...eventWrapperProps}>
          <Component {...this.props} />  
        </EventWrapper>
      );
    }
  }
  
    return EventProvider;
}

type ANY = any;
interface eventConsumerprops extends ANY {
    _eventItemId?: string
}

export const eventConsumer = (Component) => {
  class EventConsumer extends React.Component<eventConsumerprops, null> {
    render () {
        let { _eventItemId } = this.props;
        return (
            <EventContext.Consumer>
                {(context) => {
                    if (_eventItemId) {
                        context.eventItem = {
                            _eventItemId: _eventItemId,
                            title: '',
                            start_date: '',
                            end_date: '',
                            details: '',
                            createdUpdatedDateTime: '',
                            discontinued: false,
                        };

                        if (_eventItemId in context.eventItemList) {
                            context.eventItem = {
                                _eventItemId: _eventItemId,
                                title:                                   'title' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].title : '',
                                start_date:                         'start_date' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].start_date : '',
                                end_date:                             'end_date' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].end_date : '',
                                details:                               'details' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].details : '',
                                createdUpdatedDateTime: 'createdUpdatedDateTime' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].createdUpdatedDateTime : '',
                                discontinued:                     'discontinued' in context.eventItemList[_eventItemId] ? context.eventItemList[_eventItemId].discontinued : false,
                            };
                        }
                    }

                    return (
                        <Component {...this.props} {...{event: context}} />
                    )
                }}
            </EventContext.Consumer>
        );
    }
  }

  return EventConsumer;
}

















// const UPDATE_EVENT_ITEM = 'UPDATE_EVENT_ITEM';
// const updateEventItem = function(_new, _old, callback) {
//     let payload = {_id: _old._eventItemId};
//     let vaildKeys = ['title', 'start_date', 'end_date', 'details', 'discontinued'];

//     for (let i = 0; i < vaildKeys.length; i++) {
//         if (_new[vaildKeys[i]] != _old[vaildKeys[i]]) {
//             payload[vaildKeys[i]] = _new[vaildKeys[i]];
//         }
//     }
    
//     return function(dispatch) {
//         const url = '/api/event';
//         return fetch(url, {
//             method: 'PUT',
//             credentials: 'same-origin',
//             headers: {
//                 'Content-Type': 'application/json',
//                 'Accept': 'application/json',
//             },
//             body: JSON.stringify(payload), 
//         }).then(function(response) {
//             if (response.status < 200 || response.status >= 300) {
//                 const error = new Error(response.statusText)
//                 error.response = response
//                 throw error;
//             }
//             return response;
//         })
//         .then(function(response) {
//             callback(false);
//             return dispatch(updateEventItemSuccess());
//         })
//         .catch(function(error) {
//             callback(true);
//             return dispatch(updateEventItemError(error));
//         });
//     };
// };

// const UPDATE_EVENT_ITEM_SUCCESS = "UPDATE_EVENT_ITEM_SUCCESS";
// const updateEventItemSuccess = function() {
//     return {
//         type: UPDATE_EVENT_ITEM_SUCCESS,
//     };
// };

// const UPDATE_EVENT_ITEM_ERROR = 'UPDATE_EVENT_ITEM_ERROR';
// const updateEventItemError = function(error) {
//     return {
//         type: UPDATE_EVENT_ITEM_ERROR,
//         error: error,
//     };
// };

// exports.updateEventItem = updateEventItem;
// exports.UPDATE_EVENT_ITEM_SUCCESS = UPDATE_EVENT_ITEM_SUCCESS;
// exports.updateEventItemSuccess = updateEventItemSuccess;
// exports.UPDATE_EVENT_ITEM_ERROR = UPDATE_EVENT_ITEM_ERROR;
// exports.updateEventItemError = updateEventItemError;