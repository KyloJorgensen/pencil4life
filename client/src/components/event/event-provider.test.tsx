'use strict';

import * as React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import { EventWrapper, EventContext, eventProvider, eventConsumer, eventProviderOptions, IEventContext } from './event-provider';

let wrapper: ShallowWrapper | ReactWrapper;

class Elem extends React.Component<null, null> {
	render() {
		return (
			<p></p>
		)
	}
}

describe('Render EventContext', () => {
	it('matches the snapshot', () => {
		wrapper = mount(
			<div>
				<EventContext.Consumer>
					{context => {
						return (
							<p></p>
						);
					}}
				</EventContext.Consumer>
			</div>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('should have a p', () => {
		wrapper = mount(
			<div>
				<EventContext.Consumer>
					{context => {
						return (
							<p></p>
						);
					}}
				</EventContext.Consumer>
			</div>
		);
		const child = wrapper.find('p');
		expect(child.length).toEqual(1);
	});

	it('should have a p that contains "child"', () => {
		wrapper = mount(
			<div>
				<EventContext.Consumer>
					{context => {
						return (
							<p>child</p>
						);
					}}
				</EventContext.Consumer>
			</div>
		);
		const child = wrapper.find('p').contains('child');
		expect(child).toEqual(true);
	});
});

describe('Render Event component', () => {
	it('matches the snapshot', () => {
		wrapper = shallow(<EventWrapper />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should have a p', () => {
		wrapper = shallow(
			<EventWrapper>
				<p>child</p>
			</EventWrapper>
		);
		const child = wrapper.find('p');
		expect(child.length).toEqual(1);
	});

	it('should have a p that contains "child"', () => {
		wrapper = shallow(
			<EventWrapper>
				<p>child</p>
			</EventWrapper>
		);
		const child = wrapper.find('p').contains('child');
		expect(child).toEqual(true);
	});
});

describe('Render eventProvider', () => {
	it('matches the snapshot', () => {
		const EP = eventProvider(Elem);
		wrapper = shallow(<EP />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('Render eventConsumer', () => {
	it('matches the snapshot', () => {
		const EC = eventConsumer(Elem);
		wrapper = shallow(<EC />);
		expect(wrapper).toMatchSnapshot();
	});
});	

describe('evntConsumer with item id passed as prop', () => {
	beforeEach(() => {
		interface ItemProps {
			event: IEventContext;
		}

		class Item extends React.Component<ItemProps> {
			render() {
				return (
					<div>
						<p className="_eventItemId" >{this.props.event.eventItem._eventItemId}</p>
						<p className="title">{this.props.event.eventItem.title}</p>
						<p className="start_date">{this.props.event.eventItem.start_date}</p>
						<p className="end_date">{this.props.event.eventItem.end_date}</p>
						<p className="details">{this.props.event.eventItem.details}</p>
						<p className="createdUpdatedDateTime">{this.props.event.eventItem.createdUpdatedDateTime}</p>
						<p className="discontinued">{this.props.event.eventItem.discontinued? "true" : "false"}</p>
					</div>
				);
			}
		}

		const EC = eventConsumer(Item);

		class Test extends React.Component {
			render() {
				return (
					<EC _eventItemId={'12345'} />
				);
			}
		}

		const eventProvideroptions: eventProviderOptions = {
			props: {
				initialState: {
					eventItemList: {
						"12345": {
							_eventItemId: "12345",
							title: 'blah',
							start_date: '0',
							end_date: '1',
							details: 'stuff',
							createdUpdatedDateTime: 'date',
							discontinued: false,
						}
					},
					eventItems: ['12345'],
					limit: 10,
					page: 1,
					pageTotal: 1,
					total: 1,
					totalPages: 1,
				}
			}
		};


		const EP = eventProvider(Test, eventProvideroptions)

		wrapper = mount(<EP />);
	});

	it('should pass the id along', () => {
		expect(wrapper.find('p._eventItemId').text()).toEqual('12345');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.title').text()).toEqual('blah');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.start_date').text()).toEqual('0');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.end_date').text()).toEqual('1');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.details').text()).toEqual('stuff');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.createdUpdatedDateTime').text()).toEqual('date');			
	})
	it('should pass the id along', () => {
		expect(wrapper.find('p.discontinued').text()).toEqual('false');			
	})
});