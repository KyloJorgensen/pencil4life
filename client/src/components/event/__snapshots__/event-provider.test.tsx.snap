// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Event component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventWrapper
    initialState={
      Object {
        "eventItemList": Object {},
        "eventItems": Array [],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "value": Object {
        "addEventItem": [Function],
        "eventItemList": Object {},
        "eventItems": Array [],
        "getEventItem": [Function],
        "getEventItems": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
        "_currentValue2": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
        "_defaultValue": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "value": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "addEventItem": [Function],
            "eventItemList": Object {},
            "eventItems": Array [],
            "getEventItem": [Function],
            "getEventItems": [Function],
            "limit": 0,
            "page": 0,
            "pageTotal": 0,
            "total": 0,
            "totalPages": 0,
          },
          "_currentValue2": Object {
            "addEventItem": [Function],
            "eventItemList": Object {},
            "eventItems": Array [],
            "getEventItem": [Function],
            "getEventItems": [Function],
            "limit": 0,
            "page": 0,
            "pageTotal": 0,
            "total": 0,
            "totalPages": 0,
          },
          "_defaultValue": Object {
            "addEventItem": [Function],
            "eventItemList": Object {},
            "eventItems": Array [],
            "getEventItem": [Function],
            "getEventItems": [Function],
            "limit": 0,
            "page": 0,
            "pageTotal": 0,
            "total": 0,
            "totalPages": 0,
          },
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render EventContext matches the snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <div>
    <Context.Consumer>
      [Function]
    </Context.Consumer>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <div>
      <p />
    </div>,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Context.Consumer>
        [Function]
      </Context.Consumer>,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": <p />,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": Array [
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <div>
        <p />
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Context.Consumer>
          [Function]
        </Context.Consumer>,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <p />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render eventConsumer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventConsumer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_changedBits": 0,
      "_changedBits2": 0,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": Object {
        "addEventItem": [Function],
        "eventItemList": Object {},
        "eventItems": Array [],
        "getEventItem": [Function],
        "getEventItems": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      },
      "_currentValue2": Object {
        "addEventItem": [Function],
        "eventItemList": Object {},
        "eventItems": Array [],
        "getEventItem": [Function],
        "getEventItems": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      },
      "_defaultValue": Object {
        "addEventItem": [Function],
        "eventItemList": Object {},
        "eventItems": Array [],
        "getEventItem": [Function],
        "getEventItems": [Function],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
        "_currentValue2": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
        "_defaultValue": Object {
          "addEventItem": [Function],
          "eventItemList": Object {},
          "eventItems": Array [],
          "getEventItem": [Function],
          "getEventItems": [Function],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render eventProvider matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Elem />,
      "initialState": Object {
        "eventItemList": Object {},
        "eventItems": Array [],
        "limit": 0,
        "page": 0,
        "pageTotal": 0,
        "total": 0,
        "totalPages": 0,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Elem />,
        "initialState": Object {
          "eventItemList": Object {},
          "eventItems": Array [],
          "limit": 0,
          "page": 0,
          "pageTotal": 0,
          "total": 0,
          "totalPages": 0,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
